Looks like we need to make the PHAR executable after downloading it:

```
run-phar-specific-tests:

BUILD FAILED
/home/runner/work/phpunit/phpunit/build.xml:418: Execute failed: java.io.IOException: Cannot run program "/home/runner/work/phpunit/phpunit/build/artifacts/phpunit-snapshot.phar" (in directory "/home/runner/work/phpunit/phpunit"): error=13, Permission denied
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -210,8 +210,53 @@ jobs:
       - name: Assert that git tree is clean
         run: git diff || echo "Run 'php build/scripts/generate-global-assert-wrappers.php' to regenerate global assert wrappers!"
 
-  build-and-test-phar:
-    name: Build and test PHAR
+  build-phar:
+    name: Build PHAR
+
+    runs-on: ubuntu-latest
+
+    env:
+      PHP_EXTENSIONS: none, dom, json, fileinfo, iconv, libxml, mbstring, pdo_sqlite, phar, soap, tokenizer, xml, xmlwriter
+      PHP_INI_VALUES: assert.exception=1, phar.readonly=0, zend.assertions=1
+
+    strategy:
+      matrix:
+        php-version:
+          - "8.1"
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install PHP with extensions
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php-version }}
+          coverage: none
+          extensions: ${{ env.PHP_EXTENSIONS }}
+          ini-values: ${{ env.PHP_INI_VALUES }}
+          tools: none
+
+      - name: Install java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+
+      - name: Build PHAR
+        run: ant phar-snapshot
+
+      - name: Upload PHAR
+        uses: actions/upload-artifact@v2
+        with:
+          name: phpunit-snapshot-phar
+          path: ./build/artifacts/phpunit-snapshot.phar
+          retention-days: 7
+
+  test-phar:
+    name: Test PHAR
+
+    needs:
+      - build-phar
 
     runs-on: ubuntu-latest
 
@@ -246,12 +291,14 @@ jobs:
         with:
           java-version: 1.8
 
-      - name: Build PHAR and run the PHAR-specific tests with it
-        run: ant build-phar-and-run-phar-specific-tests
-
-      - uses: actions/upload-artifact@v2
-        if: ${{ matrix.php-version == 8.1 }}
+      - name: Download PHAR
+        uses: actions/download-artifact@v2
         with:
           name: phpunit-snapshot-phar
-          path: ./build/artifacts/phpunit-snapshot.phar
-          retention-days: 7
+          path: ./build/artifacts/
+
+      - name: Make PHAR exexutable
+        run: chmod +x ./build/artifacts/phpunit-snapshot.phar
+
+      - name: Run PHAR-specific tests
+        run: ant run-phar-specific-tests

